client/my-sites/checkout/composite-checkout/lib/submit-wpcom-transaction.ts

```diff
@@ -17,8 +17,13 @@ export default async function submitWpcomTransaction(
 	payload: WPCOMTransactionEndpointRequestPayload,
 	transactionOptions: PaymentProcessorOptions
 ): Promise< WPCOMTransactionEndpointResponse > {
-	if ( transactionOptions.createUserAndSiteBeforeTransaction || payload.cart.is_jetpack_checkout ) {
-		const createAccountOptions = payload.cart.is_jetpack_checkout
+	const isJetpackUserLessCheckout =
+		payload.cart.is_jetpack_checkout && payload.cart.cart_key === 'no-user';
+
+	if ( transactionOptions.createUserAndSiteBeforeTransaction || isJetpackUserLessCheckout ) {
```

Comment:
Users going through any of the new Jetpack flows can be logged in, so only checking whether `is_jetpack_checkout` is true is not enough to decide if a user account should be created.